back/engines/commercial/user_custom_fields/app/services/user_custom_fields/age_counter.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module UserCustomFields
+  class AgeCounter
+    Result = Struct.new(:binned_counts, :unknown_age_count, :bins)
+
+    DEFAULT_BINS = [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, nil].freeze
+
+    def count(users, bins = nil)
+      bins ||= DEFAULT_BINS
+      birthyear_custom_field = CustomField.find_by(key: 'birthyear')
+      birthyear_counts = FieldValueCounter.counts_by_field_option(users, birthyear_custom_field)
+
+      unknown_age_count = birthyear_counts.delete(FieldValueCounter::UNKNOWN_VALUE_LABEL)
+      age_counts = birthyear_counts.transform_keys { |birthyear| convert_to_age(birthyear) }
+      binned_counts = bin_data(age_counts, bins)
+
+      Result.new(binned_counts, unknown_age_count, bins)
+    end
+
+    private
+
+    def convert_to_age(birthyear, time: Time.zone.now)
+      # Since we don't known the exact birth date, we estimate it as the middle of the
+      # year: July 1st of the year of birth.
+      birth_time = Time.zone.local(birthyear, 7, 1)
+      (time - birth_time) / 1.year
+    end
+
+    # @param [Hash] counts
+    # @param [Array<Numeric,nil>] bins
+    def bin_data(counts, bins)
+      bins = bins.dup
+      bins[0] = 0 if bins.first.nil?
+      bins[-1] = Float::INFINITY if bins.last.nil?
```

Comment:
```suggestion
      bins[0] = 0 unless bins.first
      bins[-1] = Float::INFINITY unless bins.last
```
Are these boundary conditions described in tests? That is not clear to me.

Reply:
There are already quite a few test cases here: `engines/commercial/user_custom_fields/spec/services/user_custom_fields/age_counter_spec.rb:1`. Maybe I could add one with negative ages and check that they are discarded. As for upper bound, I don't see how I could use a test to show it is converted to infinity. How would you go about that?

Reply:
For the upper bound, isn't it covered by the `realistic_data` case in the test?